token/services/certifier/interactive/client.go

```diff
@@ -138,25 +138,32 @@ func (d *CertificationClient) Scan() {
 			logger.Debugf("request certification of [%v] satisfied with no error", toBeCertified)
 		}
 
-		done := false
-		for {
-			select {
-			case <-d.ctx.Done():
-				return
-			case <-time.After(2 * time.Second):
-				txid, err := d.vault.GetLastTxID()
-				if err != nil {
-					continue
-				}
-				if txid != lastTXID {
-					lastTXID = txid
-					done = true
-					break
-				}
-			}
-			if done {
-				break
+		nextTxId := <-d.getNextCommittedTxID(lastTXID)
+		if !nextTxId.ok {
+			return
+		}
+		lastTXID = nextTxId.value
+	}
+}
+
+func (d *CertificationClient) getNextCommittedTxID(lastTXID string) <-chan channelResult {
```

Comment:
while I really like channels, not sure why we use one here :)

we are not running `getNextCommittedTxID` in a new go routine, actually what we want is a blocking behaviour, right?

So we could just return (string, bool) and use it like that:

```go
nextTxId, ok := d.getNextCommittedTxID(lastTXID)
if !ok {
    return
}
```

In that case there is also no need for `channelResult`

Reply:
I think that we want to run the outer for-loop once per new TX ID. So we need to block until the vault gives us a new TX ID value. We could still skip the channel within the method `getNextCommittedTxID` in order to return `(string, bool)` but I think this would make the inner for loop run non stop (now we poll every 2 seconds for the result).